pyls/plugins/jedi_rename.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020 Palantir Technologies, Inc.
+import logging
+
+from pyls import hookimpl, uris, _utils
+
+log = logging.getLogger(__name__)
+
+
+@hookimpl
+def pyls_rename(config, workspace, document, position, new_name):  # pylint: disable=unused-argument
+    log.debug('Executing rename of %s to %s', document.word_at_position(position), new_name)
+    kwargs = _utils.position_to_jedi_linecolumn(document, position)
+    kwargs['new_name'] = new_name
+    refactoring = document.jedi_script().rename(**kwargs)
```

Comment:
What does Jedi do when `rename` is called for Python 2.7 or 3.5?

Reply:
It raises ``NotImplementedError``. Should we wrap that exception into a nicer one?

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/jedi/api/__init__.py", line 60, in wrapper
    "No support for refactorings/search on Python 2/3.5"
NotImplementedError: No support for refactorings/search on Python 2/3.5
```

Reply:
Yes please, let's catch the error.

Reply:
I agree with @goanpeca, Please catch the error and make this method return an empty dict with the right content so it can't processed by any client.

Reply:
What do we want to tell clients? I think this should return an error and pyls_jsonrpc already has logic to convert exceptions into rpc errors that clients should interpret as documented on the protocol. Returning an empty dict means the server will fail silently and clients will have no idea of why, if we keep the exception clients will see an error that says "No support for refactorings/search on Python 2/3.5".

We could wrap that message to say "no support for renaming on Python 2/3.5", but we should still raise an exception.

Reply:
Pushed a better error message. Let me know what you think. Here's an example with coc.nvim:

<img width="999" alt="image" src="https://user-images.githubusercontent.com/108725/81518752-125a7f00-930d-11ea-80f9-f14416d64396.png">

Reply:
I think that's quite good, thanks!